xdsl/ir/core.py

```diff
@@ -1042,6 +1070,39 @@ def last(self) -> Operation | None:
         return self.block.last_op
 
 
+@dataclass
+class BlockReverseOps:
+    """
+    Multi-pass iterable of the operations in a block. Follows the prev_op for
+    each operation.
+    """
+
+    block: Block
+
+    def __iter__(self):
+        return _BlockOpsReverseIterator(self.last)
+
+    def __len__(self):
+        result = 0
+        for _ in self:
+            result += 1
+        return result
+
+    @property
+    def last(self) -> Operation | None:
+        """
+        Last operation in the block, None if block is empty.
+        """
+        return self.block.last_op
```

Comment:
This is tricky, I would expect `iterable.last` to be the last element in the sequence, so `block.first_op` here. Maybe we'd be better off omitting this property on this class?

Reply:
That's a great observation. I dropped this for now.